### PR TITLE
add universal capability and state

### PR DESCRIPTION
**Pull request by slashmili**
@nonninz so it turns out to be a simple change! Diogo and I paired on this and will create a separate PR to explain how to add a new capability 🦾 

I have one question regarding typespec for VIN and brand. is there anything I can use or should I create one?

BTW the property tests for set/get command are really nice! I remember it was hard to debug but now the error message are much more readable.

### CODE REVIEW

File: lib/auto_api/states/universal_state.ex [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+  @moduledoc """
+  Universal state
+  """
+
+  alias AutoApi.State
+
+  use AutoApi.State, spec_file: "universal.json"
+
+  @type t :: %__MODULE__{
+          vin: State.property(String.t()), # TODO: any spec of VIn?
```
Comment (nonninz) [inline lib/auto_api/states/universal_state.ex:33]: We could add a `@type vin::<<_::17*8>>`?

File: lib/auto_api/states/universal_state.ex [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+          brand: State.property(:atom) # TODO: any spec for brand?
```
Comment (nonninz) [inline lib/auto_api/states/universal_state.ex:34]: 1. We could add a `@type brand::atom()`, to make it short.
2. We could specify the brands themselves manually:
```
@type brand::
:unknown |
:abarth |
:alfaromeo |
:alpine |
:audi |
:bmw |
:cadillac |
:chevrolet |
:chrysler |
:citroen |
:cupra |
:dacia |
:dodge |
:ds |
:fiat |
:ford |
:honda |
:hyundai |
:iveco |
:jaguar |
:jeep |
:kia |
:lancia |
:land_rover |
:lexus |
:man |
:mazda |
:mercedes_benz |
:mini |
:mitsubishi |
:nissan |
:opel |
:peugeot |
:porsche |
:renault |
:seat |
:skoda |
:smart |
:subaru |
:toyota |
:volkswagen |
:volvo_cars |
:emulator
```

3. We can somehow add it with some metaprogramming, but I don't have the metaprogramming skills to do it right now. 😬 If you have any idea how, `AutoApi.UniversalProperties.brand/0` returns the list of brand atoms as read by the specification file

**Review by nonninz — APPROVED** [review]
Overall looks great to me 👍🏻 

Added some comments on the Typespecs

File: lib/auto_api/states/universal_state.ex (hunk 2 above)
Comment (slashmili) [reply lib/auto_api/states/universal_state.ex]: Done with some Macro magics:

```
@type brand :: unquote(Enum.reduce(brands, &{:|, [], [&1, &2]}))
```

**Comment by slashmili** [comment]
@nonninz could you take a look at the macro I created for brand typespec? 

cc @scudelletti

File: lib/auto_api/universal_properties_meta.ex [hunk 3]
```diff
@@ -33,13 +33,22 @@ defmodule AutoApi.UniversalProperties.Meta do
...
+
     base_functions =
       quote do
         @properties unquote(Macro.escape(properties))
                     |> Enum.map(fn prop ->
                       {prop["id"], String.to_atom(prop["name"])}
                     end)
 
+        @type vin :: <<_::138>>
+        @type brand :: unquote(Enum.reduce(brands, &{:|, [], [&1, &2]}))
```
[the remark this answers is not in the document]
Comment (slashmili) [reply lib/auto_api/universal_properties_meta.ex:50]: We should thankful of google 😬

**Pull request merged**